framework/filters/RateLimiter.php

```diff
@@ -84,6 +91,10 @@ public function beforeAction($action)
             $this->user = Yii::$app->getUser()->getIdentity(false);
         }
 
+        if ($this->user instanceof Closure) {
+            $this->user = call_user_func($this->user, $action);
```

Comment:
```suggestion
            $this->user = $this->user($action);
```

Reply:
@samdark are you sure? `call_user_func` is commonly used in a lot of Yii framework code in this siutations.

https://github.com/yiisoft/yii2/blob/da447fd1bb9778b8b76994f3e3e14f9748c67625/framework/widgets/ListView.php#L148
https://github.com/yiisoft/yii2/blob/da447fd1bb9778b8b76994f3e3e14f9748c67625/framework/validators/RangeValidator.php#L97

But if you like, i can change it.

Reply:
OK, let it be `call_user_func`...